ts/features/pn/utils/index.ts

```diff
@@ -55,25 +56,28 @@ export const isPNOptInMessage = (
           cta2HasServiceNavigationLink:
             !!ctas.cta_2 && isServiceDetailNavigationLink(ctas.cta_2.action)
         })),
-        O.map(
-          ctaNavigationLinkInfo =>
-            ({
-              isPNOptInMessage:
-                ctaNavigationLinkInfo.cta1HasServiceNavigationLink ||
-                ctaNavigationLinkInfo.cta2HasServiceNavigationLink,
-              ...ctaNavigationLinkInfo
-            } as PNOptInMessageInfo)
-        )
+        O.map(ctaNavigationLinkInfo => ({
+          isPNOptInMessage:
+            ctaNavigationLinkInfo.cta1HasServiceNavigationLink ||
+            ctaNavigationLinkInfo.cta2HasServiceNavigationLink,
+          ...ctaNavigationLinkInfo
+        }))
       )
     ),
-    O.getOrElse(
-      () =>
-        ({
-          isPNOptInMessage: false,
-          cta1HasServiceNavigationLink: false,
-          cta2HasServiceNavigationLink: false
-        } as PNOptInMessageInfo)
-    )
+    O.getOrElse<PNOptInMessageInfo>(() => ({
+      isPNOptInMessage: false,
+      cta1HasServiceNavigationLink: false,
+      cta2HasServiceNavigationLink: false
+    }))
+  );
+
+export const isPNOptInMessage = (serviceId: ServiceId, state: GlobalState) =>
```

Comment:
What if in the future the opt-in service will be used to send a non-opt-in message? In this case, this selector will report it as an opt-in message, regardless of the fact that the message does not have the opt-in CTAs